articles/qos_deadline_liveliness_lifespan.md

```diff
@@ -0,0 +1,235 @@
+If a user wants to hook their code into these additional QoS settings then they would need to get a reference to the rmw implementation and program against the vendor specific API.
+Without the abstraction layer provided by ROS their code becomes less portable.
+
+## New QoS Settings
+
+As users start to build more robust applications, they are going to need more control over when data is being delivered and information about when the system is failing.
+To address these needs it was proposed that we start by adding support for the following new DDS QoS settings.
+
+### Deadline
+
+The deadline policy establishes a contract for the amount of time allowed between messages.
+For Subscriptions it establishes the maximum amount of time allowed to pass between receiving messages.
+For Publishers it establishes the maximum amount of time allowed to pass between sending messages.
+For Service Servers it establishes the maximum amount of time allowed to pass between receiving a request and when a response for that request is sent.
+For Service Clients it establishes the maximum amount of time allowed to pass between sending a request and when a response for that request is received.
+
+Topics and Services will support the following levels of deadlines:
+- DEADLINE_DEFAULT - Use the ROS specified default for deadlines (which is DEADLINE_NONE).
+- DEADLINE_NONE - No deadline will be offered or requested and events will not be generated for missed deadlines.
+- DEADLINE_RMW - The rmw layer of ROS will track the deadline. For a Publisher or Subscriber this means that a deadline will be considered missed if the rmw layer has not received a message within the specified time. For a Service this means the time a request is started is marked when the request reaches the rmw layer and the time at which it is finished is when the response message reaches the rmw layer.
+
+In order for a Subscriber to listen to a Publisher's Topic the deadline that they request must be greater than, or equal to, the deadline set by the Publisher.
+A Service Client will **not** be prevented from making a request to a Service Server if the  Server provides a deadline greater than the deadline requested by the Client.
+
+### Liveliness
+
+The liveliness policy establishes a contract for how entities report that they are still alive. 
+For Subscriptions it establishes the level of reporting that they require from the Publishers to which they are subscribed. 
+For Publishers it establishes the level of reporting that they will provide to Subscribers that they are alive. 
+For Service Servers it establishes both the level of reporting that they will provide to Clients and also the level of reporting that they require from Clients.
+For Service Clients it establishes both the level of reporting that they require from Service Servers and the level of reporting that they will provide to the Server.
```

Comment:
In [DDS terms](https://community.rti.com/static/documentation/connext-dds/5.2.0/doc/api/connext_dds/api_cpp2/group__DDSLivelinessQosModule.html) liveliness is defined as "Specifies and configures the mechanism that allows dds::sub::DataReader entities to detect when dds::pub::DataWriter entities become disconnected or "dead."

So above really implies that there is a 1-1 relationship between ROS 2 Publishers and DataWriters, ROS 2 Subscriptions and DataReaders and for the Servers and Clients I am not really sure.

Reply:
While it may be true, I don't think that the stated definition requires a 1-1 relationship. If for some reason an rmw implementation decided it wanted a second DataWriter/DataReader for every ROS Publisher/Subscriber then they could still do that and the definition for the ROS Pub/Sub being alive would just depend on both DataWriters being alive. 

I think this 1-1 relationship has cropped up in a few comments now, but I'm not understanding the concern about it. Do you see some problem if the 1-1 assumption is made or not made?